pydantic/json_schema.py

```diff
@@ -0,0 +1,768 @@
+        self, schema: core_schema.TupleVariableSchema | core_schema.TuplePositionalSchema
+    ) -> JsonSchemaValue:
+        json_schema: JsonSchemaValue = {'type': 'array'}
+
+        if 'mode' not in schema:
+            json_schema['items'] = {}
+
+        elif schema['mode'] == 'variable':
+            if 'items_schema' in schema:
+                json_schema['items'] = self.generate_inner(schema['items_schema'])
+            self.update_with_validations(json_schema, schema, self.ValidationsMapping.array)
+
+        elif schema['mode'] == 'positional':
+            json_schema['minItems'] = len(schema['items_schema'])
+            prefixItems = [self.generate_inner(item) for item in schema['items_schema']]
+            if prefixItems:
+                json_schema['prefixItems'] = prefixItems
+            if 'extra_schema' in schema:
+                json_schema['items'] = self.generate_inner(schema['extra_schema'])
+            else:
+                json_schema['maxItems'] = len(schema['items_schema'])
+
+        else:
+            raise ValueError(f'Unknown tuple schema mode: {schema["mode"]}')
+
+        self.update_with_validations(json_schema, schema, self.ValidationsMapping.array)
+        return json_schema
+
+    def set_schema(self, schema: core_schema.SetSchema) -> JsonSchemaValue:
+        return self._common_set_schema(schema)
+
+    def frozenset_schema(self, schema: core_schema.FrozenSetSchema) -> JsonSchemaValue:
+        return self._common_set_schema(schema)
+
+    def _common_set_schema(self, schema: core_schema.SetSchema | core_schema.FrozenSetSchema) -> JsonSchemaValue:
+        items_schema = {} if 'items_schema' not in schema else self.generate_inner(schema['items_schema'])
+        json_schema = {'type': 'array', 'uniqueItems': True, 'items': items_schema}
+        self.update_with_validations(json_schema, schema, self.ValidationsMapping.array)
+        return json_schema
+
+    def generator_schema(self, schema: core_schema.GeneratorSchema) -> JsonSchemaValue:
+        items_schema = {} if 'items_schema' not in schema else self.generate_inner(schema['items_schema'])
+        json_schema = {'type': 'array', 'items': items_schema}
+        self.update_with_validations(json_schema, schema, self.ValidationsMapping.array)
+        return json_schema
+
+    def dict_schema(self, schema: core_schema.DictSchema) -> JsonSchemaValue:
+        values_schema = self.generate_inner(schema['values_schema']).copy() if 'values_schema' in schema else {}
+        values_schema.pop('title', None)  # don't give a title to the additionalProperties
+
+        json_schema: JsonSchemaValue = {'type': 'object'}
+        if values_schema:  # don't add additionalProperties if it's empty
+            json_schema['additionalProperties'] = values_schema
+
+        self.update_with_validations(json_schema, schema, self.ValidationsMapping.object)
+        return json_schema
+
+    source_class_to_schema: tuple[tuple[Any, dict[str, Any]], ...] = (
```

Comment:
surely instead of doing, we should set `format` in `metadata` when creating the core-schema.

Reply:
that... seems like an improvement :)